File: code_samples/dart/networking_socket_programming.dart
// Networking: Socket Programming
// AI/ML Training Sample

class SocketProgramming {
  String _data = '';
  
  void process(String input) {
    _data = input;
  }
  
  String getData() => _data;
  
  bool validate() => _data.isNotEmpty;
}

void main() {
  final instance = SocketProgramming();
  instance.process('example');
  print('Data: ${instance.getData()}');
  print('Valid: ${instance.validate()}');
}

File: code_samples/elixir/networking_socket_programming.ex
# Networking: Socket Programming
# AI/ML Training Sample

defmodule SocketProgramming do
  defstruct data: ""
  
  def new(), do: %SocketProgramming{}
  
  def process(%SocketProgramming{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%SocketProgramming{data: data}), do: data
  
  def validate(%SocketProgramming{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = SocketProgramming.new()
updated = SocketProgramming.process(instance, "example")
IO.puts("Data: " <> SocketProgramming.get_data(updated))
IO.puts("Valid: " <> to_string(SocketProgramming.validate(updated)))
